src/controllers/LpaOverviewController.js

```diff
@@ -0,0 +1,37 @@
+import performanceDbApi from '../services/performanceDbApi.js' // Assume you have an API service module
+
+const LpaOverviewController = {
+  async getOverview (req, res, next) {
+    try {
+      const lpa = req.params.lpa
+
+      const response = await performanceDbApi.getLpaOverview(lpa) // Make API request
+      const data = response.data
+
+      const datasets = Object.entries(data.datasets).map(([key, value]) => {
+        return { ...value, slug: key }
+      })
+      const totalDatasets = datasets.length
+      const datasetsWithEndpoints = datasets.filter(item => item.endpoint !== null).length
```

Comment:
We can get those three counts doing way less: going over the collection once and without creating three new intermediate collections.